File: DheerajKumar/problem15.cpp
// WAP to take input 3 numbers and print the 2nd largest number among them.

#include <iostream>
using namespace std;

int secondlargest(int number1, int number2, int number3){
    
    int largest = max(number1, max(number2, number3));
    int smallest = min(number1, min(number2, number3));
    
    return (number1 + number2 + number3) - smallest - largest;
}

int main(){
    int number1, number2, number3;

    cout<<"\n\nEnter 1st number -> ";
    cin>>number1;
    cout<<"Enter 2nd numebr -> ";
    cin>>number2;
    cout<<"Enter 3nd numebr -> ";
    cin>>number3;

    cout<<"\n\n2nd Largest number -> "<< secondlargest(number1, number2, number3);
}

File: LovelyKumari/BasicProgramming/Problem15.java
import java.util.Scanner;

public class Problem15 {
    //WAP to take input 3 numbers and print the 2nd largest number among them.
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        System.out.print("Enter the first number: ");
        int num1 = sc.nextInt();
        
        System.out.print("Enter the second number: ");
        int num2 = sc.nextInt();
        
        System.out.print("Enter the third number: ");
        int num3 = sc.nextInt();

        int largest ;
        int secondLargest ;
        
        //Find second largest
        if(num1>num2 && num1>num3){
            largest=num1 ;
            secondLargest =(num2>num3) ? num2 : num3 ;

        } else if( num2 > num1 && num2 > num3){
            largest=num2 ;
            secondLargest=(num1>num3) ? num1 : num3 ;
        } else {
            largest = num3 ;
            secondLargest =(num1>num2) ? num1 : num2 ;
        }
        System.out.println("The second largest number = " + secondLargest);

    }
    
}
